install/tools/istio_vm_common.sh

```diff
@@ -0,0 +1,223 @@
+  annotations:
+    cloud.google.com/load-balancer-type: "internal"
+  labels:
+    istio: pilot
+spec:
+  type: LoadBalancer
+  ports:
+  - port: 8080
+    protocol: TCP
+  selector:
+    istio: pilot
+EOF
+cat <<EOF | kubectl apply -n kube-system -f -
+apiVersion: v1
+kind: Service
+metadata:
+  name: dns-ilb
+  annotations:
+    cloud.google.com/load-balancer-type: "internal"
+  labels:
+    k8s-app: kube-dns
+spec:
+  type: LoadBalancer
+  ports:
+  - port: 53
+    protocol: UDP
+  selector:
+    k8s-app: kube-dns
+EOF
+
+cat <<EOF | kubectl apply  $NS -f -
+apiVersion: v1
+kind: Service
+metadata:
+  name: mixer-ilb
+  annotations:
+    cloud.google.com/load-balancer-type: "internal"
+  labels:
+    istio: mixer
+spec:
+  type: LoadBalancer
+  ports:
+  - port: 9091
+    protocol: TCP
+  selector:
+    istio: mixer
+EOF
+
+cat <<EOF | kubectl apply $NS -f -
+apiVersion: v1
+kind: Service
+metadata:
+  name: istio-ca-ilb
+  annotations:
+    cloud.google.com/load-balancer-type: "internal"
+  labels:
+    istio: istio-ca
+spec:
+  type: LoadBalancer
+  ports:
+  - port: 8060
+    protocol: TCP
+  selector:
+    istio: istio-ca
+EOF
+}
+
+# Generate a kubedns file and a cluster.env
+# Parameters:
+# - name of the k8s cluster.
+function istioGenerateClusterConfigs() {
+   local K8SCLUSTER=${1}
+
+   local NS=""
+   if [[ ${ISTIO_NAMESPACE:-} != "" ]]; then
+     NS="-n $ISTIO_NAMESPACE"
+   fi
+
+  # Multiple tries, it may take some time until the controllers generate the IPs
+  for i in {1..10}
+  do
+    PILOT_IP=$(kubectl get $NS service istio-pilot-ilb -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+    ISTIO_DNS=$(kubectl get -n kube-system service dns-ilb -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+    MIXER_IP=$(kubectl get $NS service mixer-ilb -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+    CA_IP=$(kubectl get $NS service istio-ca-ilb -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+
+    if [ ${PILOT_IP} == "" -o  ${PILOT_IP} == "" -o ${MIXER_IP} == "" ] ; then
+        echo Waiting for ILBs
+        sleep 5
+    else
+        break
+    fi
+  done
+
+  if [ ${PILOT_IP} == "" -o  ${PILOT_IP} == "" -o ${MIXER_IP} == "" ] ; then
+    echo "Failed to create ILBs"
+    exit 1
+  fi
+
+  #/etc/dnsmasq.d/kubedns
+  echo "server=/svc.cluster.local/$ISTIO_DNS" > kubedns
+  echo "address=/istio-mixer/$MIXER_IP" >> kubedns
+  echo "address=/mixer-server/$MIXER_IP" >> kubedns
+  echo "address=/istio-pilot/$PILOT_IP" >> kubedns
+  echo "address=/istio-ca/$CA_IP" >> kubedns
+
+  CIDR=$(gcloud container clusters describe ${K8SCLUSTER} $(_istioGcloudOpt) --format "value(servicesIpv4Cidr)")
+  echo "ISTIO_SERVICE_CIDR=$CIDR" > cluster.env
+}
+
+
+# Install required files on a VM and run the setup script.
+#
+# Must be run for each VM added to the cluster
+# Params:
+# - name of the VM - used to copy files over.
+# - service account to be provisioned (defaults to istio.default)
+function istioProvisionVM() {
+ NAME=${1}
```

Comment:
istioBootstrapServiceInstance

Reply:
how about istioBootstrapVM (or Node?) ? Service Instance is super confusing (with the actual service ), we're provisioning a machine.

Reply:
Node is good!